### PR TITLE
Add keepalive cmd

### PR DESCRIPTION
**Pull request by ngetahun**
## How to test

We need to run a golang suse container to test the keepalive cmd

```bash
> docker pull registry.suse.com/bci/golang:1.18
> docker run -it --rm -v "$PWD":/usr/src/myapp -w /usr/src/myapp registry.suse.com/bci/golang:1.18
# inside container
> git config --global --add safe.directory /usr/src/myapp && make

# First case, unregisted system 
> ./out/suseconnect --keepalive
# Should print out an error

# Second case
> ./out/suseconnect -r C6B55B64D68F9997 && ./out/suseconnect --keepalive
# should print a success message

```

### CODE REVIEW

File: suseconnect/suseconnect.go [hunk 1]
```diff
@@ -140,6 +142,9 @@ func connectMain() {
 		output, err := connect.GetProductStatuses("json")
 		exitOnError(err)
 		fmt.Println(output)
+	} else if keepAlive {
+		err := connect.SendKeepAlivePing()
+		exitOnError(err)
```
Comment (skazi0) [inline suseconnect/suseconnect.go:147]: To match https://github.com/SUSE/connect/pull/471/files#diff-e763b73e4093dc04b49fd09cecd63710e0502fa8ccc5b71c9eb64d0723ab0445R87 you would need to add custom error handling block inside `exitOnError()`. There is a `switch` for specific error values where you can add your part for `ErrPingFromUnregistered`, print custom message and exit with wanted code. If you don't do that, `default` case will handle the error, print `SUSEConnect error: <error as string>` and exit with `1`.

**Review by skazi0 — CHANGES_REQUESTED** [review]

**Review by djoreilly — CHANGES_REQUESTED** [review]

File: suseconnect/suseconnect.go (hunk 1 above)
Comment (djoreilly) [reply suseconnect/suseconnect.go:147]: done

File: suseconnect/connectUsage.txt [hunk 2]
```diff
@@ -35,6 +35,8 @@ Manage subscriptions at https://scc.suse.com
                              format.
         --status-text        Get current system registration status in text
                              format.
+        --keepalive          Send a keepalive call to the registration server, so it can
```
[the remark this answers is not in the document]
Comment (djoreilly) [reply suseconnect/connectUsage.txt]: done

File: internal/connect/client.go [hunk 3]
```diff
@@ -200,6 +200,18 @@ func UpdateSystem(distroTarget, instanceDataFile string) error {
 	return updateSystem(sysInfoBody)
 }
 
+// send keepalive ping to SCC
+func SendKeepAlivePing() error {
+	if !IsRegistered() {
+		return ErrPingFromUnregistered
+	}
+	err := updateSystem([]byte{})
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (djoreilly) [reply internal/connect/client.go]: changed

**Review by mssola — APPROVED** [review]

**Pull request merged**